core/cap-0063.md

```diff
@@ -0,0 +1,447 @@
+this endpoint can be implemented by calling the captive-core [`getledgerentry` HTTP endpoint](#getledgerentry-endpoint).
+
+### Concerns regarding Automatic Restoration with Full State Archival
+
+In protocol 20, there was no technical reason against enabling automatic restoration. However, the proof system
+for full state archival was not finalized, and the explicit `RestoreFootprintOp` provided more future flexibility.
+Additionally, it was important for developers to build preflighting transactions into their dApp flows.
+
+If Soroban had launched with automatic restoration, it is possible many developers would attempt to avoid preflight
+by creating transaction footprint and resource declarations manually with reasonable fee and resource assumptions.
+In a system where explicit entry restoration is not required, the burden of resource discoverability is low, so
+developers are not incentivized or required to build preflight into their tx submission system. Long term this is
+a significant issue, as the addition of full state archival with proofs makes manual resource and footprint
+declaration largely impossible. Any system that did not use RPC preflight would be broken by full state archival.
+
+With this proposal, it is vital to ensure that automatic restore does not result in a situation where large numbers
+of developers circumvent preflight. Two specific features of this proposal help ensure developers are incentivized
+to build preflight into their flows now such that they are not broken later by archival proofs. Specifically,
+no in-memory read fees incentivize developers to minimize fees by using preflight to detect the maximum
+number of entries currently live. Additionally, the requirement of `evictedSorobanEntries` vectors makes manually
+constructing footprints challenging, as the developer would need to be able to distinguish live vs evicted state
+to minimize fees. Given that preflighting will result in lower on-chain fees, and given that resource footprints
+are somewhat challenging to create without captive-core seems to indicate that developers will preflight transactions
+prior to submission as intended.
+
+## Security Concerns
+
+### Memory Based DOS Attack
+
+By distinguishing between disk backed and memory backed reads at the protocol level, it is now required that
+validators maintain all live Soroban state in memory. A significant increase in live Soroban state could
+lead to an OOM based attack on the network.
+
+However, this is not a significant concern. The `sorobanLiveStateTargetSizeBytes` (see [CAP-0062](cap-0062.md))
+provides a soft cap on the amount of live Soroban state that can exist at any given time. This prevents any
+sort of runaway memory issue. Additionally, because of the eviction process, there is a natural back pressure
+applied to in-memory state, the rate at which can be controlled via network config settings. Finally, the rate
```

Comment:
Maybe we should highlight this on top that background eviction will remove in-memory Soroban entries.